ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -2675,6 +2685,61 @@ impl<'a> Assemble<'a> {
         dynasm!(self.asm ; cmovnc Rq(true_reg.code()), Rq(false_reg.code()));
     }
 
+    fn cg_select_float(&mut self, iidx: jit_ir::InstIdx, inst: &jit_ir::SelectInst) {
+        let ([cond_reg], [true_reg, false_reg, out_reg]) = self.ra.assign_regs(
+            &mut self.asm,
+            iidx,
+            [GPConstraint::Input {
+                op: inst.cond(self.m),
+                in_ext: RegExtension::Undefined,
+                force_reg: None,
+                clobber_reg: false,
+            }],
+            [
+                RegConstraint::Input(inst.trueval(self.m)),
+                RegConstraint::Input(inst.falseval(self.m)),
+                RegConstraint::Output,
+            ],
+        );
+        debug_assert_eq!(
+            self.m
+                .type_(inst.cond(self.m).tyidx(self.m))
+                .bitw()
+                .unwrap(),
+            1
+        );
+
+        assert_eq!(
+            inst.trueval(self.m).bitw(self.m),
+            inst.falseval(self.m).bitw(self.m)
+        );
+        match inst.trueval(self.m).bitw(self.m) {
+            32 => {
+                dynasm!(self.asm
+                    ;   bt Rd(cond_reg.code()), 0
```

Comment:
Would the `fcmov` instruction help here? That would bring this in parity with the integer version.

Reply:
You tell me. I ended up baffled by this; the documentation on this stuff is much sparser -- and frankly, much worse -- than for general purpose registers. Personally I'd be happy to get something we're confident is *correct* in and optimise it later.

Reply:
Now I look at it, `fcmov` uses the old fashioned floating point stack and not the xmm registers, so we can't (and shouldn't) use it.

Your codegen looks correct to me. AI tells me there is no xmm equivalent to `fcmov` and gave me a technique similar to what you have done here.

Reply:
Sounds like we can merge this then?